test/dubhash.test.ts

```diff
@@ -0,0 +1,108 @@
+import { promises } from 'fs';
+import * as path from 'path';
+import { SignatureResult } from '../lib/types';
+import { getDubHashSignature } from '../lib/dubhash';
+const { readFile } = promises;
+
+describe('getDubHashSignature', () => {
+  it('returns correct SignatureResult for binary file', async () => {
+    const fixturePath = path.join(__dirname, 'fixtures');
+    const filePath = path.join(fixturePath, 'dubhash', 'test02');
+    const fileContents = await readFile(filePath);
+    const expected: SignatureResult = {
+      path: filePath,
+      hashes_ffm: [
+        {
+          format: 1,
+          data: 'aT6a+E09/MceZA4AW9xeLg', //TODO: Verify with Daniel
+        },
+      ],
+    };
+    const actual = getDubHashSignature(filePath, fileContents);
+    expect(actual?.hashes_ffm[0].data.length).toEqual(22);
+    expect(actual).toStrictEqual(expected);
+  });
+
+  it('returns correct SignatureResult for text file - both hashes', async () => {
+    const fixturePath = path.join(__dirname, 'fixtures');
+    const filePath = path.join(fixturePath, 'dubhash', 'test00');
+    const fileContents = await readFile(filePath);
+    const expected: SignatureResult = {
+      path: filePath,
+      hashes_ffm: [
+        {
+          format: 1,
+          data: 'Bp5by58MFpH4B4WNuaY6IA',
```

Comment:
not that it matters much, but we could have read the test00.result, and parsed it into an object
by doing so we could have the same code for testing all the tests

Reply:
I see your point, but I prefer to have the expected hashes "in line" in the Typescript. In my opinion, it makes the tests a bit more readable.